### PR TITLE
Just Implement Cache for get Table ID based on table name

### PR DESCRIPTION
**Pull request by yamelsenih**
This pull equest implement cache for **MTable.getTable_ID(String tableName)** method

### CODE REVIEW

File: base/src/org/compiere/model/MTable.java [hunk 1]
```diff
@@ -876,22 +878,13 @@ public String getSQLCreate()
...
-			rs = pstmt.executeQuery();
-			if (rs.next())
-				retValue = rs.getInt(1);
-		} catch (Exception e) {
-			s_log.log(Level.SEVERE, SQL, e);
-			retValue = -1;
-		} finally {
-			DB.close(rs, pstmt);
+		Integer tableId = s_tableNameCache.get(tableName);
+		if(Optional.ofNullable(tableId).orElse(new Integer(0)) > 0) {
```
Comment (mckayERP) [inline base/src/org/compiere/model/MTable.java:882]: If using the Optional pattern, you could use .OrElseGet and do it all in one line

    public static int getTable_ID(String tableName) {

        return Optional.ofNullable(s_tableNameCache.get(tableName))
                .orElseGet(() -> 
                    getTable_IDAndAddToCache(tableName)
                );

    }

    private static Integer getTable_IDAndAddToCache(String tableName) {

        int retValue = DB.getSQLValue(null,
                "SELECT AD_Table_ID FROM AD_Table WHERE tablename = ?",
                tableName);
        s_tableNameCache.put(tableName, retValue);
        return retValue;

    }

Would the transaction name ever be required here?

**Review by mckayERP — APPROVED** [review]
Added a suggestion but its not required.  Good improvement.

File: base/src/org/compiere/model/MTable.java (hunk 1 above)
Comment (e-Evolution) [reply base/src/org/compiere/model/MTable.java:882]: @mckayERP is right is not same orElse that  orElseGet to runtime the first is evaluate and second only if get None

**Comment by yamelsenih** [comment]
Hi @mckayERP already exists your recommendations

**Comment by mckayERP** [comment]
I'm OK with it.  Thanks.

**Pull request merged**